packages/adapter/lib/adapter/adapter.js

```diff
@@ -8731,6 +8734,53 @@ function Adapter(options) {
     this.pluginHandler = new PluginHandler(pluginSettings);
     this.pluginHandler.addPlugins(this.ioPack.common.plugins, [this.adapterDir, __dirname]); // first resolve from adapter directory, else from js-controller
 
+    /**
+     * This method returns the list of license that can be used by this adapter
+     * @param {boolean} all if return the licenses, that used by other instances (true) or only for this instance (false)
+     * @returns {object[]} list of suitable licenses
+    */
+    this.getSuitableLicenses = async all => {
+        const licenses = [];
+        try {
+            const obj = await this.getForeignObjectAsync('system.licenses');
+            if (obj && obj.native && obj.native.licenses && obj.native.licenses.length) {
+                const now = Date.now();
+                const cert = fs.readFileSync(__dirname + '/../../cert/cloudCert.crt');
+                const version = this.pack.version.split('.')[0];
```

Comment:
we could also use `semver.major` for this.

Reply:
semver does not now in adapter.js, so I didn't want to add new dependency. Especially it could be solved so easy

Reply:
Yes, I've just seen that it is added in pack.json, but it seems to be unused, probably an artefact of the refactoring